File: src/paradigms/objectoriented/Airport.java
package paradigms.objectoriented;

import java.util.ArrayList;
import java.util.List;

public class Airport {
    
    public String name;
    public List<Flight> flights = new ArrayList<>();

    public Airport(String name) {
        this.name = name;
    }

    void addFlight(Flight flight) {
        flights.add(flight);
    }

    public static void main(String[] args) {
        Airport airport = new Airport("Melbourne");
        Flight flight = new Flight("A224", "SF");
        Passenger harry = new Passenger("P2402", "Harry");

        airport.addFlight(flight);
        flight.boardPassenger(harry);
        System.out.printf(
            "%s is now at %s\n", harry.getName(), harry.getLocation()
        );

        flight.depart();
        System.out.printf(
            "%s is now at %s\n", harry.getName(), harry.getLocation()
        );
    }
}

File: src/paradigms/procedural/airport.ts
/* airport as a series of processes operating on data structures */

import * as Model from './model.ts';

function airportProcedural() {
    const bookings: Model.BookingDatabase = {
        passengers: [],
        flights: []
    };

    function addPassenger(id: string, name: string, flightID: string) {
        bookings.passengers.push({id, name, flightID});

        const flight = bookings.flights.find(f => f.id == flightID);

        if (flight) {
            flight.passenger.push(id);
        }
    }

    function registerDeparture(flightID: string) {
        const index = bookings.flights.findIndex(f => f.id == flightID);

        if (index != -1) {
            bookings.flights.splice(index, 1);
            bookings.passengers = bookings.passengers.filter(
                (p) => p.flightID == flightID
            );
        }
    }

    return { addPassenger, registerDeparture, bookings };
}

const proc = airportProcedural();
proc.addPassenger('P2402', 'Harry', 'A224');

console.log('Passengers:', proc.bookings.passengers);

export { airportProcedural };